### ISSUE TITLE
Gradle 3.0 support

### ISSUE THREAD

**Issue by msgilligan** [asciidoctor/asciidoclet#62]
I'm trying to upgrade my [bitcoinj-addons](https://github.com/msgilligan/bitcoinj-addons) repository to Gradle 3.1-rc-1 and when I try to run Gradle (even a `gradle clean`) I get the following error message:

```
Cannot change strategy of configuration ':asciidoclet' after it has been resolved.
```

This is true as of https://github.com/msgilligan/bitcoinj-addons/commit/e22470eb3b56c92dd7ff488f6a7ff91ea313bc14

Am I doing something wrong? Has anyone used _asciidoclet_ with Gradle 3.x yet?

p.s. the "Composite Builds" feature of Gradle 3.1 is really awesome -- people with library projects are going to want to upgrade to it quickly, I think.

**Comment by johncarl81** [comment]
@msgilligan, I'm not seeing this error message when I build from master.  Is it happening on another branch perhaps?

**Comment by msgilligan** [comment]
You're building from master with `./gradlew` or with a locally installed Gradle 3.0 and `gradle`?

**Comment by msgilligan** [comment]
And I should also mention that I get deprecation warnings under 2.14.1 indicating that the build will break under 3.0...

**Comment by johncarl81** [comment]
Looks like I missed these updates.. im building using `./gradlew clean build`, which uses version 2.14.1.  Using the non-wrapper gradle from a manual install (https://gradle.org/gradle-download/) I get the following:
```
gradle clean build
Parallel execution is an incubating feature.
Using the 'clean' task in combination with parallel execution may lead to unexpected runtime behavior.

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'bitcoinj-addons'.
> Cannot change strategy of configuration ':asciidoclet' after it has been resolved.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED
```

.. Digging into it further.

**Comment by johncarl81** [comment]
@benevans, would you mind taking a look at this?  I may be out of my depth with Gradle.

**Comment by msgilligan** [comment]
Any updates on this issue? It's keeping me from upgrading beyond Gradle 2.x and I might have to revert to standard JavaDoc from AsciiDoc in order to upgrade Gradle.

**Comment by johncarl81** [comment]
My appologies @msgilligan, I have not focused on this issue.  @benevans, any chance you could look at this?  @mojavelinux?

**Comment by johncarl81** [comment]
@msgilligan, I was able to build your project after upgrading gradle to version 3.1.  Could you walk me through reproducing the error?

**Comment by msgilligan** [comment]
Yeah, seems to be working for me now, too. It fails if you check out the exact commit I referenced in the initial report. I'm guessing the upgrade to Groovy or another dependency upgrade I made had a side effect of fixing this issue.

I think we can go ahead and close it and reopen (or open a new one) if the problem raises its head again.

I just [upgraded](https://github.com/msgilligan/bitcoinj-addons/commit/eddae9859012bcfba7951646d80992d7191957e9) the `master` branch of `bitcoinj-addons` all the way to the latest Gradle, version 3.5.
